persist/fs/files.go

```diff
@@ -53,6 +53,9 @@ func openFiles(opener fileOpener, fds map[string]**os.File) error {
 
 func closeFiles(fds ...*os.File) error {
 	for _, fd := range fds {
+		if fd == nil {
```

Comment:
I see why now, as you use this in the error case for opening them... Yeah this is probably fine.  If we wanted it to be safer we could make the first parameter `allowNilFiles` or something like that and we pass it true when we call it in the case of files opening causing an error.

Or we could filter them out by adding another method:
```go
func validFiles(fds ...*os.File) []*os.File {
  var result []*os.File
  for _, fd := range fds {
    if fd != nil {
      result = append(result, fd)
    }
  }
  return result
}

// Call with:
closeFiles(validFiles(myFile1, myFile2, myFile3))
```